test.js

```diff
@@ -2,85 +2,198 @@
+                beforeEach(function(){
+                    csvtojsonStream = gulpCsvtojson({ globalvariable: 'gv', genjs: true });
+                });
+
+                it("should return js when globalvariable is set null",function(done){                    
+                    csvtojsonStream.once('data',function(file){
+                        assert(file.contents.toString(),  "gv=" + expected + ";" );
+                        done();
+                    });
+
+                    csvtojsonStream.write(file2);
+                    csvtojsonStream.end();
+                });
+
+            });
+        });
+
+        describe("When using a Stream",function(){
+            var file = null;
+            beforeEach(function(){
+                file = new File({
+                    path : 'any.csv',
+                    contents : stringToStream(csvData)
+                });
+            });
+            describe("When using default options",function(){
+                var csvtojsonStream = null;
+                beforeEach(function(){
+                    csvtojsonStream = gulpCsvtojson();
+                });
+
+                it("should generate expected json string",function(done){
+                    var expectedResult = [{"Header1":"col1-row1","Header2":"col2-row1","Header3":"col3-row1"},{"Header1":"col1-row2","Header2":"col2-row2","Header3":"col3-row2"}];
+                    csvtojsonStream.on('data',function(file){
+                        file.contents.pipe(concat(function(data){
+                            assert(JSON.stringify(JSON.parse(data.toString())) == JSON.stringify(expectedResult));
+                            done();
+                        }));
+                    });
+                    csvtojsonStream.write(file);
+                    csvtojsonStream.end();
+                });
+                it("should rename path to a json file",function(done){
+                    csvtojsonStream.on('data',function(file){
+                        assert.equal(file.relative,'any.json');
+                        done();
+                    });
+
+                    csvtojsonStream.write(file);
+                    csvtojsonStream.end();
+                });
+            });
+
+            describe("When defining generate js option",function(){
+                var csvtojsonStream = null;
+                beforeEach(function(){
+                    csvtojsonStream = gulpCsvtojson({genjs:true});
+                });
+
+                it("should generate expected json string",function(done){
+
+                    var expectedResult = 'var any = [{"Header1":"col1-row1","Header2":"col2-row1","Header3":"col3-row1"},{"Header1":"col1-row2","Header2":"col2-row2","Header3":"col3-row2"}];';
+                    csvtojsonStream.on('data',function(file){
+                        file.contents.pipe(concat(function(data){
+                            assert(data.toString().replace(/\s\n/g,'') == expectedResult);
```

Comment:
suggest to use `assert.equal` which produce better result (output the expected and actual value) when failed.